### DISCUSSION TITLE
SignalR and Redis across multiple environments

### DISCUSSION THREAD

**Discussion by slovely** [dotnet/aspnetcore#29149] [Q&A]
I have an application that uses SignalR with the `Microsoft.AspNetCore.SignalR.Redis` nuget package.  If I install the application in multiple environments (dev, test, staging, etc), is it safe for them all to use the same redis server without interference?  I am not passing in anything special in the configuration:
```            
var redisConnectionStrings = Configuration.GetConnectionString("RedisSignalrServer");
            if (!string.IsNullOrWhiteSpace(redisConnectionStrings))
                signalrBuilder.AddRedis(redisConnectionStrings);
```
Is there anything else I would need to do or is that good enough?

**Comment by BrennanConroy** [comment]
> Is there anything else I would need to do or is that good enough?

It's generally a bad idea to share resources between your different environments. For example, if you introduce a bug in your dev or test environment that uses up all of the Redis CPU then your prod environment would be affected.

Having said that, if you still want to use the same Redis instance then you should add a `ChannelPrefix` to the `Configuration` that is unique per environment. That way each environment wont talk to the other environments through Redis.

**Reply by slovely** [reply]
Thank you @BrennanConroy

Agree with your comment on production - that one was always going to have its own! Just we have a fairly large number of test environments so sharing resources for them makes sense.

I will have a look at the channel prefix, thank you so much for taking the time to reply.
